### PR TITLE
Making the 'RestoreSources' configurable: CLI

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: test/NuGet.config [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (livarcocc) [inline test/NuGet.config]: Why is this here? I imagine that the tests projects should use the same RestoreSources as the source projects. Otherwise, we will end up with packages outside the build feed.

File: TestAssets/NuGet.Config [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (livarcocc) [inline TestAssets/NuGet.Config]: Same here, not sure if this is correct. That means that when we set the PackageSources from the build from sources, the TestAssets are not going to respect it.

**Comment by johnbeisner** [comment]
Replaced by: https://github.com/dotnet/cli/pull/8123
